### PR TITLE
*: make SET TRANSACTION ISOLATION LEVEL READ COMMITTED take effect

### PR DESCRIPTION
**Pull request by tiancaiamao**
it is equivalent to setting "tx_isolation" variable.
if the value is READ-COMMITTED, set the transaction isolation level option.

@disksing @zimulala @shenli

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -4836,27 +4836,72 @@ SetStmt:
...
+	}
 
 TransactionChar:
 	"ISOLATION" "LEVEL" IsolationLevel
+	{
+		tp := types.NewFieldType(mysql.TypeString)
+		tp.Charset, tp.Collate = parser.charset, parser.collation
+		expr := ast.NewValueExpr($3)
+		expr.SetType(tp)
+		$$ = &ast.VariableAssignment{Name: "tx_isolation", Value: expr, IsSystem: true}
```
Comment (shenli) [inline parser/parser.y]: Use a const for tx_isolation.

File: session.go [hunk 2]
```diff
@@ -1130,6 +1130,9 @@ func (s *session) ActivePendingTxn() error {
 	if err != nil {
 		return errors.Trace(err)
 	}
+	if s.sessionVars.Systems[variable.TxnIsolation] == "READ-COMMITTED" {
```
Comment (shenli) [inline session.go]: Use const for READ-COMMITTED.

File: parser/parser.y (hunk 1 above)
Comment (tiancaiamao) [inline parser/parser.y]: `tx_isolation` is a session variable name.
I don't want to put it in other places except variable package.
I also don't like parser package to import variable package.
So use the string "tx_isolation" is the best compromise.

**Comment by disksing** [comment]
Any need to print error when user wants to set unsupported levels?

**Comment by disksing** [comment]
Need to set `kv.Request` too.

**Comment by tiancaiamao** [comment]
```
Begin;
SET SESSION TRANSACTION ISOLATION LEVEL READ COMMITTED;
SELECT * FROM T;    // isolation level should take effect in this query.
COMMIT;
```

**Comment by tiancaiamao** [comment]
PTAL @disksing @shenli @zimulala

**Comment by disksing** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**